### ISSUE TITLE
Build errors I don't understand --- wrong number of template arguments

### ISSUE THREAD

**Issue by joelgiedt** [usqcd-software/qdpxx#36]
[make.log](https://github.com/usqcd-software/qdpxx/files/4838976/make.log)

**Comment by joelgiedt** [comment]
Compiler situation:

module load gcc/8.4.0
module load cuda/10.2
module load openmpi/4.0.3
(base) [DWFLgied@npl41 ~]$ mpicxx -v
Using built-in specs.
COLLECT_GCC=/gpfs/u/software/npl/gcc/8.4.0/1/bin/g++
COLLECT_LTO_WRAPPER=/gpfs/u/software/npl/gcc/8.4.0/1/libexec/gcc/x86_64-pc-linux-gnu/8.4.0/lto-wrapper
Target: x86_64-pc-linux-gnu
Configured with: ./configure --disable-multilib --prefix=/gpfs/u/software/npl/gcc/8.4.0/1
Thread model: posix
gcc version 8.4.0 (GCC)
(base) [DWFLgied@npl41 ~]$

**Comment by joelgiedt** [comment]
I switched from --enable-parallel-arch=parscalarvec to --enable-parallel-arch=parscalar in the configure and the error went away.  My compiler doesn't like the OSubLattice<T2,Subset> argument in copymask of qdp_parscalarvec_specific.h.

**Comment by joelgiedt** [comment]
I managed to build with --enable-parallel-arch=parscalarvec but had to make some changes to a few files.  All of the places where two template parameters to OSubLattice were given in qdp_parscalarvec_specific.h were changed to just one, OSubLattice<T> instead, to mirror what happens in qdp_parscalar_specific.h.  I also had to add "std::" in a few places where errors were occuring for references to ostream and nothrow.
